### PR TITLE
allow overwrite sitemovers queuedata settings from AGIS

### PR DESCRIPTION
**Pull request by anisyonk**
Updates:
 - implemented base functionality to parse and overwrite data passed via `jobPars` settings
 - introduced `--overwriteAGISdata` arg option to overwrite queuedata coming from AGIS
 - added `copytools` and `acopytools` entries to overwriteAGISdata, update affected stagein/out functions

The format of `overwriteAGISdata` command line option in JobParameters is a string representation of python structures (need to be overwritten)

e.g.
```
jobPars = '
--overwriteAGISdata "{\'allowfax\':False,\'copytools\':{\'rucio\':{\'setup\':\'fakesetup.sh\'}, \'acopytools\':{\'pr\':[\'rucio\']}}"
'
```

to simply overwrite queuedata both for stage-in and stage-out (e.g. by HC jobs) and force use of `Rucio` sitemover following settings can be passed:
```
--overwriteAGISdata "{'copytools':{'rucio':{'setup':''}, 'acopytools':{}}"
```

### CODE REVIEW

File: Job.py [hunk 1]
```diff
@@ -682,6 +684,60 @@ def setJobDef(self, data):
         if not self.accessmode:
             self.accessmode = 'direct' if self.isAnalysisJob() else 'copy'
 
+        try:
+            import ast
+            self.overwriteAGISData = self.parseJobPars(self.jobPars, {'--overwriteAGISdata': lambda x: ast.literal_eval(x) if x else {}}).get('--overwriteAGISdata', {})
```
Comment (PalNilsson) [inline Job.py]: Are you removing --overwriteAGISdata from jobPars somewhere? I don't see it. If it's still there when the trf starts then the trf will fail.

**Comment by anisyonk** [comment]
actually I don't, got it. I will check today and push updates to this request.

**Comment by anisyonk** [comment]
Paul, I have updated the logic.

**Comment by PalNilsson** [comment]
Hi,

Doesn’t quite work however; if the new option is not used, then too much is removed:

2017-12-04 13:34:21|17710|Job.py      | INFO: parseJobPars(): do extract options=['--overwriteAGISdata'] from data=--maxEvents=1 --inputHITSFile HITS.06828093._000096.pool.root.1 --outputRDOFile RDO_405f1da7-47c5-4aec-a97d-5c18e59cd264.root
2017-12-04 13:34:21|17710|Job.py      | job.overwriteAGISData={}
2017-12-04 13:34:21|17710|Job.py      | cleaned job.jobPars=HITS.06828093._000096.pool.root.1 --outputRDOFile RDO_405f1da7-47c5-4aec-a97d-5c18e59cd264.root

Cheers, Paul

**Pull request merged**